File: 0853-car-fleet/0853-car-fleet.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:

        new_array = sorted(zip(position,speed),reverse= True)
        stack = []
        for pos,spd in new_array:
            time =(target-pos) / spd 
            if not stack or time > stack[-1]:
                stack.append(time)
        return len(stack)

File: 0853-car-fleet/0853-car-fleet.java
class Solution {
    public int carFleet(int target, int[] position, int[] speed) {
        int n = position.length;
        Integer[] idx = new Integer[n];
        for (int i = 0; i < n; i++) idx[i] = i;

        // Sort indices by position descending
        Arrays.sort(idx, (a, b) -> position[b] - position[a]);

        int fleets = 0;
        double maxTime = 0;

        for (int i : idx) {
            double time = (double)(target - position[i]) / speed[i];
            if (time > maxTime) {   // new fleet
                maxTime = time;
                fleets++;
            }
            // else: merges with fleet ahead, skip
        }

        return fleets;
    }
}